ext/mysql2/client.c

```diff
@@ -1086,6 +1086,41 @@ static VALUE rb_mysql_client_encoding(VALUE self) {
 #endif
 
 /* call-seq:
+ *    client.automatic_close?
+ *
+ * @return [Boolean]
+ */
+static VALUE get_automatic_close(VALUE self) {
+  GET_CLIENT(self);
+  return wrapper->automatic_close ? Qtrue : Qfalse;
+}
+
+/* call-seq:
+ *    client.automatic_close = false
+ *
+ * Set this to +false+ to leave the connection open after it is garbage
+ * collected. To avoid "Aborted connection" errors on the server, explicitly
+ * call +close+ when the connection is no longer needed.
+ *
+ * @see http://dev.mysql.com/doc/en/communication-errors.html
+ */
+static VALUE set_automatic_close(VALUE self, VALUE value) {
+  GET_CLIENT(self);
+  if (RTEST(value)) {
+    value = Qtrue;
+    wrapper->automatic_close = 1;
+  } else {
+#ifndef _WIN32
+    value = Qfalse;
+    wrapper->automatic_close = 0;
+#else
+    rb_raise(cMysql2Error, "Connections are always closed by garbage collector on Windows");
```

Comment:
I would rb_warn here, rather than rb_raise. I'm imagining an app that comes with this setting baked in, and then someone wants to deploy on Windows, but ends up blocked until the code is updated.